src/ast/ast_validations.c

```diff
@@ -830,21 +833,23 @@ static AST_Validation _Validate_CREATE_Clauses(const AST *ast, char **reason) {
 
 	uint *create_clause_indices = AST_GetClauseIndices(ast, CYPHER_AST_CREATE);
 	uint clause_count = array_len(create_clause_indices);
+	rax *defined_aliases = NULL;
 	if(clause_count == 0) goto cleanup;
-	for(uint i = 0; i < clause_count; i++) {
-		const cypher_astnode_t *clause = cypher_ast_query_get_clause(ast->root, create_clause_indices[i]);
-		// Verify that functions invoked by the CREATE clause are valid.
-		res = _ValidateFunctionCalls(clause, reason, false);
-		if(res == AST_INVALID) goto cleanup;
 
-		if(_Validate_CREATE_Clause_TypedRelations(clause) == AST_INVALID) {
-			asprintf(reason, "Exactly one relationship type must be specified for CREATE");
-			res = AST_INVALID;
-			goto cleanup;
+	defined_aliases = raxNew();
+	uint start_offset = 0;
+	for(uint i = 0; i < clause_count; i ++) {
+		uint clause_idx = create_clause_indices[i];
+
+		// Collect all entities that are bound before this CREATE clause.
+		for(uint j = start_offset; j < clause_idx; j ++) {
+			const cypher_astnode_t *prev_clause = cypher_ast_query_get_clause(ast->root, i);
+			_AST_GetDefinedIdentifiers(prev_clause, defined_aliases);
 		}
+		start_offset = clause_idx;
 
-		// Validate that inlined properties do not use undefined identifiers.
-		res = _Validate_CREATE_Clause_Properties(clause, reason);
+		const cypher_astnode_t *clause = cypher_ast_query_get_clause(ast->root, clause_idx);
+		res = _Validate_CREATE_Entities(clause, defined_aliases, reason);
```

Comment:
does `defined_aliases` gets cleared between each loop iteration?

Reply:
No, each iteration adds all the aliases introduced from clauses[i-1] to clauses[i] to the same rax. We use similar logic in MERGE validation. This validation is used on scoped ASTs, so this is preferable to resetting.